Java compilation failure in relationship when using UUID as Id field type

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**
When we have an UUID as Id field in Entity A and use that Entity as part of a OneToMany relationship, there are java complation errors within domain entity, but also within RowMappers and so on. According to #13258 this is supposed to work. 

Error output during compilation (mvnw clean package -Pprod)
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.9.0:compile (default-compile) on project dataview: Compilation failure: Compilation failure:
[ERROR] /C:/Users/jwedding/git/mt/demo4/src/main/java/de/mt/dataview/domain/LogStream.java:[36,13] cannot find symbol
[ERROR]   symbol:   class UUID
[ERROR]   location: class de.mt.dataview.domain.LogStream
[ERROR] /C:/Users/jwedding/git/mt/demo4/src/main/java/de/mt/dataview/domain/LogStream.java:[106,12] cannot find symbol
[ERROR]   symbol:   class UUID
[ERROR]   location: class de.mt.dataview.domain.LogStream
[ERROR] /C:/Users/jwedding/git/mt/demo4/src/main/java/de/mt/dataview/domain/LogStream.java:[110,34] cannot find symbol
[ERROR]   symbol:   class UUID
[ERROR]   location: class de.mt.dataview.domain.LogStream

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

##### **Motivation for or Use Case**
Backend doesn't compile. 
<!-- Explain why this is a bug for you -->

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
Generate a application where a OneToMany relationship contains an UUID field as Id in the "One" side of the relationship.

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->
#17823 

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->
I'll try to open a PR by myself again, fixing solely the compilation issue for now.

##### **JHipster Version(s)**

<!--
Which version of JHipster are you using, is it a regression?
-->
7.6.0

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->


##### **JHipster Version(s)**

```
dataview@0.0.1-SNAPSHOT C:\Users\jwedding\git\mt\demo4
`-- (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "oauth2",
    "baseName": "dataview",
    "blueprints": [],
    "buildTool": "maven",
    "cacheProvider": "no",
    "clientFramework": "react",
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1639736978010,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "devServerPort": 9060,
    "dtoSuffix": "DTO",
    "enableGradleEnterprise": false,
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "entities": ["DataStream", "LogStream"],
    "entitySuffix": "",
    "incrementalChangelog": true,
    "jhiPrefix": "jhi",
    "jhipsterVersion": "7.6.0",
    "languages": ["de", "en"],
    "lastLiquibaseTimestamp": 1644393955000,
    "messageBroker": false,
    "nativeLanguage": "de",
    "otherModules": [],
    "packageName": "de.mt.dataview",
    "pages": [],
    "prodDatabaseType": "mysql",
    "reactive": true,
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "searchEngine": false,
    "serverPort": "8080",
    "serverSideOptions": [],
    "serviceDiscoveryType": "no",
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "skipJhipsterDependencies": true,
    "skipUserManagement": true,
    "testFrameworks": [],
    "websocket": false,
    "withAdminUi": true
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity DataStream {
  @Id uuid UUID
}
entity LogStream {
  comment String
  status String
  date String
}
relationship OneToMany {
  DataStream{logStreams} to LogStream{dataStreams}
}

dto * with mapstruct
paginate DataStream, LogStream with pagination
service DataStream, LogStream with serviceImpl

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.12" 2021-07-20 LTS
OpenJDK Runtime Environment 18.9 (build 11.0.12+7-LTS)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.12+7-LTS, mixed mode)

git version 2.35.1.windows.2

node: v14.18.2

npm: 6.14.15

Docker version 20.10.9, build c2ea9bc


No change to package.json was detected. No package manager install will be executed.
Congratulations, JHipster execution is complete!


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/main/java/package/domain/Entity.java.jhi.ejs
+++ b/generators/entity-server/templates/src/main/java/package/domain/Entity.java.jhi.ejs
@@ -69,7 +69,7 @@ import java.time.Duration;
 import java.util.HashSet;
 import java.util.Set;
 <%_ } _%>
-<%_ if (fieldsContainUUID) { _%>
+<%_ if (fieldsContainUUID || otherEntityPrimaryKeyTypesIncludesUUID) { _%>
 import java.util.UUID;
 <%_ }
 Object.keys(uniqueEnums).forEach(function(element) { _%>

--- a/generators/entity-server/templates/src/main/java/package/repository/rowmapper/EntityRowMapper.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/repository/rowmapper/EntityRowMapper.java.ejs
@@ -28,7 +28,7 @@ import java.time.LocalDate;
 import java.time.ZonedDateTime;
 <%_ } if (fieldsContainDuration) { _%>
 import java.time.Duration;
-<%_ } if (fieldsContainUUID) { _%>
+<%_ } if (fieldsContainUUID || otherEntityPrimaryKeyTypesIncludesUUID) { _%>
 import java.util.UUID;
 <%_ } _%>
 import java.util.function.BiFunction;

